tools/updater/main.c

```diff
@@ -934,174 +862,220 @@ static k4a_result_t command_inspect_firmware(updater_command_info_t *command_inf
+            }
         }
         else
         {
-            printf("The firmware has been updated to the following firmware Versions:\n");
-            printf("  RGB camera firmware:    %d.%d.%d => %d.%d.%d\n",
-                   command_info->current_version.rgb.major,
-                   command_info->current_version.rgb.minor,
-                   command_info->current_version.rgb.iteration,
-                   command_info->updated_version.rgb.major,
-                   command_info->updated_version.rgb.minor,
-                   command_info->updated_version.rgb.iteration);
-            printf("  Depth camera firmware:  %d.%d.%d => %d.%d.%d\n",
-                   command_info->current_version.depth.major,
-                   command_info->current_version.depth.minor,
-                   command_info->current_version.depth.iteration,
-                   command_info->updated_version.depth.major,
-                   command_info->updated_version.depth.minor,
-                   command_info->updated_version.depth.iteration);
-            printf("  Depth config file:      %d.%d => %d.%d\n",
-                   command_info->current_version.depth_sensor.major,
-                   command_info->current_version.depth_sensor.minor,
-                   command_info->updated_version.depth_sensor.major,
-                   command_info->updated_version.depth_sensor.minor);
-            printf("  Audio firmware:         %d.%d.%d => %d.%d.%d\n",
-                   command_info->current_version.audio.major,
-                   command_info->current_version.audio.minor,
-                   command_info->current_version.audio.iteration,
-                   command_info->updated_version.audio.major,
-                   command_info->updated_version.audio.minor,
-                   command_info->updated_version.audio.iteration);
+            printf("ERROR: Failed to get updated versions\n\n");
+        }
+
+        // Ensure the final handle state is copied back.
+        command_info->firmware_handle = sub_command_info.firmware_handle;
+        command_info->firmware_serial_number = sub_command_info.firmware_serial_number;
+
+        printf("\n\n");
+
+        if (K4A_FAILED(result))
+        {
+            // keep track of overal status
+            finalCmdStatus = K4A_RESULT_FAILED;
         }
-    }
-    else
-    {
-        printf("ERROR: Failed to get updated versions\n\n");
     }
 
-    return result;
+    return finalCmdStatus;
 }
 
 static k4a_result_t command_reset_device(updater_command_info_t *command_info)
 {
-    k4a_result_t result = ensure_firmware_open(command_info);
-    if (!K4A_SUCCEEDED(result))
-    {
-        return result;
-    }
+    k4a_result_t finalCmdStatus = K4A_RESULT_SUCCEEDED;
+    k4a_result_t result;
 
-    result = firmware_reset_device(command_info->firmware_handle);
-    if (!K4A_SUCCEEDED(result))
+    for (uint32_t device_index = 0; device_index < command_info->device_count; device_index++)
```

Comment:
Agreed, I think we should just serialize the calls here so that we complete the reset before we go on to the next one.